task.rb

```diff
@@ -0,0 +1,128 @@
+class Station
+  def initialize(name)
+    @name = name
+    @trains = []
+  end
+
+  def take_train(train)
+    @trains << train
+  end
+
+  def trains_list
+    @trains.each{|train| puts train}
+  end
+
+  def trains_list_type(type)
+    @trains.each{|train| puts train if train.type == type}
+  end
+
+  def send_train(train_name)
+    @trains.select!{|train| train.name != train_name}
+  end
+end
+
+
+class Route
+  attr_reader :stations
+
+  def initialize(from, to)
+    @from = from
+    @to = to
+    @iterim_stations = []
+    @stations = [@from, @to]
+  end
+
+  def update_stations
+    @stations = [@from, *@iterim_stations, @to];
+  end
+
+  def add_station(station)
+    @iterim_stations << station
+    update_stations
+  end
+
+  def remove_station(station)
+    @iterim_stations.select!{|elem| elem != station}
+    update_stations
+  end
+
+  def print_stations
+    puts @from
+    @iterim_stations.each{|station| puts station}
+    puts @to
+  end
+end
+
+class Train
+  attr_accessor :speed
+  attr_reader :cars_count
+
+  def initialize(number, type, cars_count)
+    @number = number
+    @type = type
+    @cars_count = cars_count
+    @speed = 0
+    @current_index_station = 0
+    @route
+  end
+
+  def stop
+    @speed = 0
+  end
+
+  def add_car
+    if speed == 0
+      @cars_count += 1
+    else
+      return 'Speed is not 0, car can\'t be added'
+    end
+  end
+
+  def remove_car
+    if speed == 0
+      @cars_count -= 1
+    else
+      return 'Speed is not 0, car can\'t be removed'
+    end
+  end
+
+  def route(klass)
+    @route = klass.stations
+  end
+
+  def to_next_station
+    if @current_index_station != @route.length - 1
+      @current_index_station += 1
+    else
+      return 'Train is located in end of route'
+    end
+  end
+
+  def to_prev_station
+    if @current_index_station != 0
+      @current_index_station -= 1
+    else
+      return 'Train is located in begin of route'
+    end
+  end
+
+  def next_station
+    if @current_index_station != @route.length - 1
```

Comment:
```suggestion
    if current_station != @route.last_station
```
такой код больше "рассказывает" про то что он проверяет
он самодокументированный